packages/sprotty/src/features/routing/bezier-edge-router.ts

```diff
@@ -317,79 +317,64 @@ export class BezierEdgeRouter extends AbstractEdgeRouter {
  */
 export class BezierMouseListener extends MouseListener {
 
-    @inject(EdgeRouterRegistry) @optional() edgeRouterRegistry?: EdgeRouterRegistry;
-
     mouseDown(target: SModelElement, event: MouseEvent): Action[] {
-        if (!(target instanceof SRoutingHandle) || (target.kind !== 'bezier-add' && target.kind !== 'bezier-remove')) {
-            return [];
-        }
-
         const result = [];
-        let router;
-        const element = target.root.index.getById(target.id);
-        if (this.edgeRouterRegistry && element instanceof SRoutingHandle) {
-            router = this.edgeRouterRegistry.get((element.parent as SRoutableElement).routerKind);
-        }
-        if (router instanceof BezierEdgeRouter) {
+        if (target instanceof SRoutingHandle && (target.kind === 'bezier-add' || target.kind === 'bezier-remove')) {
             if (target.type === 'bezier-create-routing-point') {
-                result.push(new BezierCurveAction(router, BezierActionTask.ADD, target.pointIndex, target.parent.id));
+                result.push(new AddRemoveBezierSegmentAction(AddRemoveBezierSegmentActionTask.ADD, target.id));
             } else if (target.type === 'bezier-remove-routing-point') {
-                result.push(new BezierCurveAction(router, BezierActionTask.REMOVE, target.pointIndex, target.parent.id));
+                result.push(new AddRemoveBezierSegmentAction(AddRemoveBezierSegmentActionTask.REMOVE, target.id));
             }
         }
         return result;
     }
 };
 
-export enum BezierActionTask {
+export enum AddRemoveBezierSegmentActionTask {
     ADD,
     REMOVE
 }
 
-export class BezierCurveAction implements Action {
+export class AddRemoveBezierSegmentAction implements Action {
```

Comment:
Following my recent refactoring, I'd change this to an interface:
```typescript
export interface AddRemoveBezierSegmentAction extends Action {
    kind: typeof AddRemoveBezierSegmentAction.KIND
    targetId: string
    actionTask: 'add' | 'remove'
}
export namespace AddRemoveBezierSegmentAction {
    export const KIND = 'addRemoveBezierSegment';
    export function create(actionTask: 'add' | 'remove', targetId: string): AddRemoveBezierSegmentAction {
        return {
            kind: KIND,
            actionTask,
            targetId
        };
    }
}
```

Reply:
I will adapt as you suggested. Do you want to refactor all actions implementations in the future?

Reply:
Yes, actions should always be defined as interfaces.